agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/FileStringLookup.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.configuration;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+import javax.annotation.Nullable;
+import org.apache.commons.text.lookup.StringLookup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class FileStringLookup implements StringLookup {
+
+  private static final Logger logger = LoggerFactory.getLogger(FileStringLookup.class);
+  private static final String PREFIX = "${file:";
+  private static final String LESS_STRICT_PREFIX = "file:";
```

Comment:
What is the use case for supporting the LESS_STRICT_PREFIX ?

Reply:
use the `StringSubstitutor`, this will handle the templating `${...}`, and then your `FileStringLookup` doesn't need to

Reply:
it's easy to make mistake in the json config.  
here i want to support these two formats:
`${file:file.txt} ` or `file:file.txt` 

the goal here is the get the file name.  that's why I allow less strict.

Reply:
will try stringSubstitutor.

Reply:
> it's easy to make mistake in the json config. here i want to support these two formats: `${file:file.txt} ` or `file:file.txt`
> 
> the goal here is the get the file name. that's why I allow less strict.

Got it.  My recommendation is to keep it simple.  If we will not document the less strict things then I don't think we need to anticipate and support the various typos people can make.  If we will document then that is all the more reason to keeping it simple.  I'm not going to block the review on this.  I will let you make the call.

Reply:
@ramthi this was addressed by switching to standard StringSubstitutor (which only handles strict behavior)